### DISCUSSION TITLE
如何为socks和http代理添加密码认证？

### DISCUSSION THREAD

**Discussion by NiallLDY** [docker-easyconnect/docker-easyconnect#132] [Q&A]
我想把这个部署到公网服务器，暴露socks代理和http代理端口，这样是不是不大安全？
有没有为这两个代理添加用户名密码认证的功能？就类似于clash的http代理就可以添加用户名密码来认证

**Comment by Hagb** [comment]
[danted](https://www.inet.no/dante/) 和 [tinyproxy](https://github.com/tinyproxy/tinyproxy) 分别为容器提供 socks 和 http 代理，它们的配置文件位于代码仓库中的 `docker-root/etc/`，运行容器时 tinyproxy 直接使用 `/etc` 的配置文件，而 danted 使用的配置文件从 `danted.conf.sample` 生成，其中 `#external-lines` 一行由若干行 `external: IFACE` 替换（其中 `IFACE` 指 network interface 的名字）。

这两个代理软件应该都可以通过配置文件设置认证方式。（欢迎 PR）
